src/map/clif.c

```diff
@@ -14298,7 +14298,7 @@ static void clif_parse_CloseVending(int fd, struct map_session_data *sd) __attri
 /// 012e
 static void clif_parse_CloseVending(int fd, struct map_session_data *sd)
 {
-	if (pc_istrading(sd) || pc_isdead(sd))
+	if (sd->npc_id || sd->state.buyingstore || sd->state.trading)
```

Comment:
probably better leave pc_isdead? Or you think dead players should close vending too?

Reply:
The original client allows you to close it while dead (tested on 2018-10), so if we keep the pc_isdead check, we will prevent them from closing while dead (although the client would still close it, as it doesn't wait for an ACK), a player could close it while dead and when they respawn the shop would be open again (client-side wise), which is a wrong behaviour.

I haven't tried, but I think adding a check that prevents them from closing while dead would also cause issues with a priest using ressurrect, example (not tested, just an idea):
- merchant dies
- merchant clicks on close (client shows it as closed / server doesn't close it)
- priest ressurrects player
- merchant still has the vending state, thus couldn't walk, but they doesn't have an open shop in their UI -> player gets stuck = bug